### PR TITLE
WIP BXMSDOC-6064: Added traits chapter in decision engine doc as per BAPL-1625

### PR DESCRIPTION
**Pull request by hmanwani-rh**
Added **chapter 83. Traits in the decision engine** in the Developing Decision Services document

See JIRA: https://issues.redhat.com/browse/BXMSDOC-6064
Epic: https://issues.redhat.com/browse/BXMSDOC-6359
BAPL: https://issues.redhat.com/browse/BAPL-1625

Doc previews:
[Developing Decision Services in Red Hat Process Automation Manager](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-6064-RHPAM/#con-traits_decision-engine)
[Developing Decision Services in Red Hat Decision Manager](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-6064-RHDM/#con-traits_decision-engine)

### CODE REVIEW

File: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
+[id='con-traits_{context}']
+
+= Traits in the {DECISION_ENGINE}
+Traits in {DECISION_ENGINE} allow you to model multiple dynamic types that do not fit in a class hierarchy. A trait is an interface that you can apply (and eventually remove) to an individual object at runtime. However, traits are not supported in the executable rule model, therefore traits functionality and its specific implementation code is moved to an individual module named `drools-traits`. To use traits, you can import the `drools-traits` module to Maven.
+
+The `drools-traits` module is based on the `drools-core` and `drools-compiler` module. The classes from `drools-core` and `drools-compiler` are moved to the `drools-traits` module, containing `.traits` prefix in package name and `-Impl` suffix in class name to avoid the split package problem.
```
Comment (lucamolteni) [inline doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:6]: This was a technical detail to provide other engineers some details on the work done, but it's irrelevant to the end users. We can safely remove this paragraph and the next note

File: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+[source]
+----
+org.drools.core.reteoo.KieComponentFactoryFactory=org.drools.traits.core.reteoo.TraitKieComponentFactoryFactory
+org.drools.compiler.builder.impl.TypeDeclarationBuilderFactory=org.drools.traits.compiler.builder.impl.TraitTypeDeclarationBuilderFactory
+org.kie.api.persistence.jpa.KieStoreServices = org.drools.persistence.jpa.KnowledgeStoreServiceImpl
+?org.kie.internal.process.CorrelationKeyFactory = org.jbpm.persistence.correlation.JPACorrelationKeyFactory
+----
+
+IMPORTANT: Persistence related tests are moved to the `drools-traits` module. Therefore, the `kie.conf` file contains `KieStoreServices` defined for persistence.
```
Comment (lucamolteni) [inline doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:32]: This is also irrelevant, up until the end I'd say.

**Review by lucamolteni — CHANGES_REQUESTED** [review]
Some of the parts are too technical for the end users, let's remove them

**Comment by lucamolteni** [comment]
@mariofusco please take a look at it

File: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc (hunk 2 above)
Comment (hmanwani-rh) [reply doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:32]: @lucamolteni  Would you suggest keeping only the introduction paragraph and removing the rest content including components?

Comment (lucamolteni) [reply doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:32]: yes @hmanwani-rh

File: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc (hunk 1 above)
Comment (mariofusco) [reply doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:6]: I agree that this paragraph is not relevant for users and can be removed.

File: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc (hunk 2 above)
Comment (mariofusco) [reply doc-content/drools-docs/src/main/asciidoc/DecisionEngine/con-traits.adoc:32]: @lucamolteni Agreed on everything.

**Pull request closed without merging**